File: ios/RNJWPlayer/RCTConvert+RNJWPlayer.m
//
//  RCTConvert+RNJWPlayer.m
//  RNJWPlayer
//
//  Created by Chaim Paneth on 10/29/21.
//

#import "RCTConvert+RNJWPlayer.h"

@implementation RCTConvert (RNJWPlayer)

RCT_ENUM_CONVERTER(JWAdClient, (@{
    @"vast": @(JWAdClientJWPlayer),
    @"ima": @(JWAdClientGoogleIMA),
    @"ima_dai": @(JWAdClientGoogleIMADAI),
}), JWAdClientUnknown, integerValue)

RCT_ENUM_CONVERTER(JWInterfaceBehavior, (@{
    @"normal": @(JWInterfaceBehaviorNormal),
    @"hidden": @(JWInterfaceBehaviorHidden),
    @"onscreen": @(JWInterfaceBehaviorAlwaysOnScreen),
}), JWInterfaceBehaviorNormal, integerValue)

RCT_ENUM_CONVERTER(JWCaptionEdgeStyle, (@{
    @"none": @(JWCaptionEdgeStyleNone),
    @"dropshadow": @(JWCaptionEdgeStyleDropshadow),
    @"raised": @(JWCaptionEdgeStyleRaised),
    @"depressed": @(JWCaptionEdgeStyleDepressed),
    @"uniform": @(JWCaptionEdgeStyleUniform),
}), JWCaptionEdgeStyleUndefined, integerValue)

RCT_ENUM_CONVERTER(JWPreload, (@{
    @"auto": @(JWPreloadAuto),
    @"none": @(JWPreloadNone),
}), JWPreloadNone, integerValue)

RCT_ENUM_CONVERTER(JWRelatedOnClick, (@{
    @"play": @(JWRelatedOnClickPlay),
    @"link": @(JWRelatedOnClickLink),
}), JWRelatedOnClickPlay, integerValue)

RCT_ENUM_CONVERTER(JWRelatedOnComplete, (@{
    @"show": @(JWRelatedOnCompleteShow),
    @"hide": @(JWRelatedOnCompleteHide),
    @"autoplay": @(JWRelatedOnCompleteAutoplay),
}), JWRelatedOnCompleteShow, integerValue)

RCT_ENUM_CONVERTER(JWControlType, (@{
    @"forward": @(JWControlTypeFastForwardButton),
    @"rewind": @(JWControlTypeRewindButton),
    @"pip": @(JWControlTypePictureInPictureButton),
    @"airplay": @(JWControlTypeAirplayButton),
    @"chromecast": @(JWControlTypeChromecastButton),
    @"next": @(JWControlTypeNextButton),
    @"previous": @(JWControlTypePreviousButton),
    @"settings": @(JWControlTypeSettingsButton),
    @"languages": @(JWControlTypeLanguagesButton),
    @"fullscreen": @(JWControlTypeFullscreenButton),
}), JWControlTypeFullscreenButton, integerValue)

@end

File: ios/RNJWPlayer/RCTConvert+RNJWPlayer.swift
//
//  RNJWPlayerViewController.m
//  RNJWPlayer
//
//  Created by Chaim Paneth on 3/30/22.
//

import Foundation
import React
import JWPlayerKit

extension RCTConvert {

    static func JWAdClient(_ value: String) -> JWAdClient {
        switch value {
        case "vast":
            return .JWPlayer
        case "ima":
            return .GoogleIMA
        case "ima_dai":
            return .GoogleIMADAI
        default:
            return .unknown
        }
    }

    static func JWInterfaceBehavior(_ value: String) -> JWInterfaceBehavior {
        switch value {
        case "normal":
            return .normal
        case "hidden":
            return .hidden
        case "onscreen":
            return .alwaysOnScreen
        default:
            return .normal
        }
    }

    static func JWCaptionEdgeStyle(_ value: String) -> JWCaptionEdgeStyle {
        switch value {
        case "none":
            return .none
        case "dropshadow":
            return .dropshadow
        case "raised":
            return .raised
        case "depressed":
            return .depressed
        case "uniform":
            return .uniform
        default:
            return .undefined
        }
    }

    static func JWPreload(_ value: String) -> JWPreload {
        switch value {
        case "auto":
            return .auto
        case "none":
            return .none
        default:
            return .none
        }
    }

    static func JWRelatedOnClick(_ value: String) -> JWRelatedOnClick {
        switch value {
        case "play":
            return .play
        case "link":
            return .link
        default:
            return .play
        }
    }

    static func JWRelatedOnComplete(_ value: String) -> JWRelatedOnComplete {
        switch value {
        case "show":
            return .show
        case "hide":
            return .hide
        case "autoplay":
            return .autoplay
        default:
            return .show
        }
    }

    static func JWControlType(_ value: String) -> JWControlType {
        switch value {
        case "forward":
            return .fastForwardButton
        case "rewind":
            return .rewindButton
        case "pip":
            return .pictureInPictureButton
        case "airplay":
            return .airplayButton
        case "chromecast":
            return .chromecastButton
        case "next":
            return .nextButton
        case "previous":
            return .previousButton
        case "settings":
            return .settingsButton
        case "languages":
            return .languagesButton
        case "fullscreen":
            return .fullscreenButton
        default:
            return .fullscreenButton
        }
    }

}
